File: packages/lib-backend/src/file/utils/toRelative/toRelative.ts
import { fromWorking } from '@lib/backend/file/utils/fromWorking/fromWorking';
import {
  type ToRelativeModel,
  type ToRelativeParamsModel,
} from '@lib/backend/file/utils/toRelative/toRelative.models';
import { relative } from 'path';

export const toRelative = ({ from = fromWorking(), to }: ToRelativeParamsModel): ToRelativeModel =>
  relative(from, to);

File: packages/lib-shared-py/src/lib_shared/path/utils/to_relative/to_relative.py
from os import path
from pathlib import Path

from lib_shared.path.utils.from_working import from_working


def to_relative(
    source: str,
    target: str,
) -> str:
    src = Path(source or from_working())
    tgt = Path(target)
    try:
        return str(tgt.relative_to(src))
    except ValueError:
        return str(Path(path.relpath(tgt, src)))
